src/tasks/decode.ts

```diff
@@ -0,0 +1,333 @@
+    console.log(label(`Receiver`), receiver);
+  }
+  if (appData !== constants.HashZero) {
+    console.log(label(`AppData`), appData);
+  }
+  console.log(
+    label(`Signature (${formatSignature(signingScheme)})`),
+    signature,
+  );
+}
+
+function displayInteractions(
+  interactions: [Interaction[], Interaction[], Interaction[]],
+) {
+  console.log(chalk.bold("=== Decoded interactions ==="));
+  console.log(chalk.gray("-".repeat(WIDTH)));
+  displayInteractionGroup("Pre-interactions", interactions[0]);
+  console.log();
+  displayInteractionGroup("Intra-interactions", interactions[1]);
+  console.log();
+  displayInteractionGroup("Post-interactions", interactions[2]);
+  console.log();
+}
+
+function displayInteractionGroup(name: string, interactions: Interaction[]) {
+  const nonEmpty = interactions.length !== 0;
+  console.log(` ${nonEmpty ? "┌" : " "}--- ${name} ---`);
+  for (const interaction of interactions.slice(undefined, -1)) {
+    displayInteraction(interaction, false);
+  }
+  if (nonEmpty) {
+    displayInteraction(interactions[interactions.length - 1], true);
+  }
+}
+
+function displayInteraction(interaction: Interaction, isLast: boolean) {
+  let newInteraction = true;
+  const branch = () => {
+    if (newInteraction) {
+      newInteraction = false;
+      return isLast ? " └─── " : " ├─── ";
+    } else {
+      return isLast ? "      " : " │    ";
+    }
+  };
+  const { target, value, callData } = interaction;
+  console.log(branch(), mainLabel("Interaction"), `target address ${target}`);
+  if (!BigNumber.from(value).isZero()) {
+    console.log(branch(), label("Value"), utils.formatEther(value));
+  }
+  console.log(branch(), label("Calldata"), callData);
+}
+
+const setupDecodeTask: () => void = () => {
+  task("decode", "Decodes GPv2 settlement calldata.")
+    .addOptionalParam(
+      "txhash",
+      "The transaction hash of the transaction to decode. Tf this flag is set, stdin is ignored.",
+    )
+    .setAction(
+      async ({ txhash }, { artifacts, ethers, deployments, network }) => {
+        let calldata;
+        if (txhash !== undefined) {
+          const tx = await ethers.provider.getTransaction(txhash.trim());
+          if (tx === null) {
+            throw new Error(`Transaction not found on network ${network.name}`);
+          }
+          const deployment = await deployments
+            .get("GPv2Settlement")
+            .catch(() => null);
+          calldata = tx.data;
+          if (deployment === null || tx.to !== deployment.address) {
+            console.log(
+              `Warning: the input transaction hash does not point to an interaction with the current deployment of GPv2 settlement contract on ${network.name}.`,
+            );
+            console.log(`Deployment: ${deployment?.address}`);
+            console.log(`Target:     ${tx.to}`);
+          }
+        } else {
+          calldata = (await fs.readFile("/dev/stdin")).toString().trim();
```

Comment:
nanonit: maybe print that you are waiting on user pasting the calldata (or are you expecting people to know and only use this with the `<` operator)?

Reply:
I feel that is a little "un-Unix-y" :stuck_out_tongue:

Reply:
As Nick says, this was set up as a Unix tool. :smile: 
The use case I imagined is to extract the calldata from the log and then pipe it into this command. But pasting tha calldata after running the command makes sense, I'll try to see how it looks like.